lib/repos/article_progress_repo.dart

```diff
@@ -0,0 +1,32 @@
+import 'dart:async';
+import 'package:maui/db/entity/article_progress.dart';
+import 'package:maui/db/dao/article_progress_dao.dart';
+
+class ArticleProgressRepo {
+  const ArticleProgressRepo();
+  static final ArticleProgressDao articleProgressDao = new ArticleProgressDao();
+
+  Future<String> insertArticleProgress(String id, String userId, String topicId,
+      String articleId, String timeStampId) async {
```

Comment:
Always better to have this as a named parameter. It gets confusing to the caller otherwise. Also you can make timeStampId optional and if not supplied, can call DateTime.now().millisecondsSinceEpoch